src/gl/canvas2dRe.re

```diff
@@ -0,0 +1,76 @@
+type t;
+
+/* Canvas API, following https://simon.html5.org/dump/html5-canvas-cheat-sheet.html */
+
+external save : unit = "" [@@bs.send.pipe: t];
+external restore : unit = "" [@@bs.send.pipe: t];
+
+/* Transformation */
+external scale : float => float => unit = "" [@@bs.send.pipe: t];
+external rotate : float => unit = "" [@@bs.send.pipe: t];
+external translate : float => float => unit = "" [@@bs.send.pipe: t];
+external transform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+external setTransform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+
+/* Compositing */
+external globalAlpha : t => float => unit = "" [@@bs.set];
+external globalCompositeOperation : t => string => unit = "" [@@bs.set];
```

Comment:
There's several problems with what you're doing here. The first is that `compositeT` needs to be a polymorphic variant type (see https://realworldocaml.org/v1/en/html/variants.html#polymorphic-variants). The second problem is that the type needs to be inlined in the type annotation of the external. You can't have it as a separate `type` declaration.

This makes it a bit cumbersome and limited in use. it might be better to just have a module with constants:
```reason
module Composite = {
  type t = string;

  let sourceOver : t = "source-over";
  ...
};

external globalCompositeOperation : t => Composite.t => unit = ""
```